src/data/set/intervals/basic.lean

```diff
@@ -921,7 +919,12 @@ lemma Ico_union_Ico (h₁ : min a b ≤ max c d) (h₂ : min c d ≤ max a b) :
 begin
   cases le_total a b with hab hab; cases le_total c d with hcd hcd; simp [hab, hcd] at h₁ h₂,
   { exact Ico_union_Ico' h₂ h₁ },
-  all_goals { simp [*] },
+  { simp * },
+  { simp * },
+  have : ¬min a c < max b d,
+  { simp only [min_lt_iff, lt_max_iff, not_or_distrib, not_lt],
+    exact ⟨⟨hab, h₁⟩, h₂, hcd⟩ },
+  simp [*],
```

Comment:
This makes it seem like the new lemmas are much less useful for `simp`

Reply:
Yeah... that's what I was worried about.

Reply:
Can this proof be changed back to what it was before now?